### ISSUE TITLE
On version 2.0 reference to self.id raises E1101 (no-member) 

### ISSUE THREAD

**Issue by alfonsopola** [PyCQA/pylint-django#184]
**WARNING:**  Please do not report issues about missing Django, see
[README](https://github.com/PyCQA/pylint-django#installation)!

**TODO:** make sure to post the output of `pip freeze`
appnope==0.1.0
astroid==2.0.1
backcall==0.1.0
decorator==4.3.0
Django==2.0.7
django-extensions==2.1.0
django-organizations==1.1.1
djangorestframework==3.8.2
dry-rest-permissions==0.1.10
ipython-genutils==0.2.0
isort==4.3.4
jedi==0.12.0
lazy-object-proxy==1.3.1
mccabe==0.6.1
parso==0.2.1
pexpect==4.6.0
pickleshare==0.7.4
prompt-toolkit==1.0.15
psycopg2==2.7.5
ptyprocess==0.5.2
Pygments==2.2.0
pylint==2.0.1
pylint-django==2.0
pylint-plugin-utils==0.4
pytz==2018.5
simplegeneric==0.8.1
six==1.11.0
traitlets==4.3.2
typed-ast==1.1.0
uWSGI==2.0.17.1
wcwidth==0.1.7
wrapt==1.10.11
yapf==0.22.0

**NOTES:**  make sure you have the latest version of 3rd party packages
like `rest_framework`, `factory`, `model_utils`, etc. before reporting
issues!

I have a model like this:
```
class MyModel(models.Model):
    __str__(self):
        return self.id
```

And I get the following error when running the linter:
`E1101: Instance of 'MyModel' has no 'id' member (no-member)`

The id member is automatically added by Django.

**Comment by jambonrose** [comment]
Additionally, this error is now raised on the `declared_fields` attribute of forms. For instance:

```python
from django.contrib.auth.forms import UserCreationForm

UserCreationForm.declared_fields
```
```
E1101: Class 'UserCreationForm' has no 'declared_fields' member (no-member)
```
